internal/controller/clusterdeployment_reconciler.go

```diff
@@ -94,31 +94,60 @@ func clusterInstallConditionTypes() []hivev1.ClusterDeploymentConditionType {
 	}
 }
 
-func updateCIProvisionedStatus(cd *hivev1.ClusterDeployment, sc *v1alpha1.ClusterInstance) {
-	// Check if cluster has finished installing:
-	// - if it has then update ClusterInstance.Status.Conditions.Provisioned -> Completed
-	if cd.Spec.Installed {
-		conditions.SetStatusCondition(&sc.Status.Conditions,
+func updateCIProvisionedStatus(cd *hivev1.ClusterDeployment, ci *v1alpha1.ClusterInstance, log logr.Logger) {
```

Comment:
This doesn't seem to cover (logically anyway) all the combinations of these conditions. I assume we're initializing this to some sane default then updating it only in these cases?

Reply:
Updated as per our discussion with early conditional `return` to reduce cognitive complexity and enhance readability.